### DISCUSSION TITLE
230808 스프린트 기능 확정 회의

### DISCUSSION THREAD

**Discussion by splitCoding** [woowacourse-teams/2023-shook#218] [회의]
스프린트 기능 확정 회의

**Comment by splitCoding** [comment]
# 스프린트 기능 확정

## 1. 파트에 좋아요 기능 도입

  - 각 노래의 3개의 킬링파트 모두 좋아요가 가능하다.
  - 좋아요를 다시 누르면 좋아요가 취소된다.

## 2. 노래 상세페이지에서 다른 노래 전환 기능 도입
    
위아래 방향으로 스와이프하여 이전 노래, 다음 노래로 이동한다. 

## 3. 노래 상세페이지에서 띄워질 UI 를 재정의한다.

1. 영상
2. 킬링파트 정보
  - 킬링파트를 좋아요 순으로 정렬하여 1,2,3 을 보여준다.
  - 좋아요
  - 댓글
3. 전체 듣기 버튼, 반복 듣기 스위치


|  | 목록 | 스와이프 | 
|:------:|:------:|:---:|
| 장점 | 노래를 고를 수 있는 자유도 | 킬링파트를 통해서 노래를 접할 수 있는 가치에 도움 |
| 장점 |  | 나중에 제공하는 방식을 랜덤으로 바꿔서 제공할 수 있음 |
| 장점 |  | 사용자가 간단하게 다음 노래로 이동할 수 있다. |
| 부족한 점 | 목록에서 노래를 골라야하는 행위가 추가됨 | 원하는 노래로 쉽게 건너뛸 수 없음 |

## 4. 메인페이지 킬링파트 수집 이벤트 기능

킬링파트 수집중인 노래는 메인페이지에서 캐러셀로 나타낸다.

<img width="393" alt="image" src="https://github.com/woowacourse-teams/2023-shook/assets/75476562/fa3713e3-9027-4a18-bef9-f19e43edb4f2">

노래 킬링파트 수집 이벤트 페이지에서는 스와이프 통해 다음 노래 킬링파트 수집 이벤트 페이지로 이동한다.

## 5. 메인페이지 노래 목록 좋아요 순 정렬

노래의 총 좋아요 순으로 정렬하여 메인페이지에 보여준다. ( 40개 )







## FE
## BE

**Reply by splitCoding** [reply]
# BE 기능 정리 

## 기능 목록

1. 파트 좋아요 등록, 취소 API 구현
2. 소셜 로그인 API 구현
3. 스와이프를 위한 API 구현
  - 좋아요 순으로 정렬된 노래 목록에서 앞 뒤 노래를 포함한 목록 조회 API 구현
  - 수집 중인 노래 목록에서 앞 뒤 노래를 포함한 목록 조회 API 구현
4. 노래 상세 페이지 API 수정
5. 킬링파트 수집 중인 노래 목록 조회 API 구현
6. 좋아요 순으로 정렬된 노래 목록 조회 API 구현 ( 기존 투표순 API 삭제 )

---

## DB 관련 기능 목록

### table schema 변경
1. part_like ( 파트 좋아요 테이블 )
2. vote 에 deleted column 추가 
3. voting-song ( 킬링파트 수집 중인 노래 테이블 )

### 추후 데이터 처리 방식
- 수집한 파트 데이터, 수집한 투표 데이터 : 킬링파트 수집 이벤트 종료시 soft delete
- 수집 중인 노래 데이터 : 킬링파트 수집 이벤트 종료시 hard delete ( 실제 song 테이블로 이전 )

---

## 도메인 변경 목록

1. 수집 중인 노래 : 수집 중인 노래의 파트 목록을 가지고 있다.
2. 수집 중인 노래의 파트 : 수집 중인 노래의 파트 투표 목록을 가지고 있다.
3. 등록이 완료된 노래 : 킬링파트 목록을 가지고 있다.
2. 킬링 파트 : 킬링파트 좋아요 목록을 가지고 있다.

**Comment by Creative-Lee** [comment]
## 이번주

1. 메인 페이지 케러셀, 바텀 시트 - 우코
2. 디테일 페이지 디자인 변경 사항 추가 및 좋아요 관련 기능 추가, 스와이프 기능 - 도밥
3. 유튜브 영상 재생과 관련된 추가 기능 개발 (볼륨 조절, 음소거, 영상 정지, 구간반복 기능 개선) 1번째로는 구간반복 우선 - 코난


## 다음주

1. 로그인 페이지 구현, 로그인 관련 코드 추가(api, 권한 별로 다르게 보이는 화면)
2. 마이 페이지
